common/utils.h

```diff
@@ -20,6 +20,10 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef __cplusplus
```

Comment:
Yes, it's not yet pushed. Mac build failed, I tried to fix it, found the problem and decided to solve it in this commit. I've been waiting for your response on my suggestions as well.